### PR TITLE
Logger statistics

### PR DESCRIPTION
**Pull request by SenecaAnderson**
Summary
Logger statistics display in the "Log File" dropdown in the sidebar of the result page.

Code Changes
- Changed return value of InMemoryTransport
- Changed LogList to account for new return value
- Modified sidebar to calculate number of errors, warnings, debug messages, and total messages

New Behavior
When you click on the Log File dropdown in the sidebar of the result page, the number of each type of message (excluding info-level messages) will appear in a list.

Testing Guidance
Run the app with npm start. Run extraction with any input that doesn't cause fatal errors. Click on "Log File" in the sidebar. The message counts will appear underneath.

### CODE REVIEW

File: electron/InMemoryTransport.js
Comment (jafeltra) [inline electron/InMemoryTransport.js:15]: I'm guessing you were using these `console.log`s to develop/debug, but we should remove them now.

File: react-app/src/components/ResultSidebar.js
Comment (jafeltra) [inline react-app/src/components/ResultSidebar.js:41]: I don't think we need to report the number of debug messages. The debug messages just provide additional detail, but the number of them isn't really indicative of the number of problems that will require attention, like the error and warn messages.

File: react-app/src/components/ResultSidebar.js
Comment (jafeltra) [inline react-app/src/components/ResultSidebar.js:40]: Since these `<p>` elements (and the ones below) are not in a loop, I don't think we need to have the `key` values on them. There should be no warnings in the console when you remove them.

File: react-app/src/components/ResultSidebar.js
Comment (jafeltra) [inline react-app/src/components/ResultSidebar.js:43]: Since the total count of resources was moved to the top, we should move the total messages count to match.

**Review by jafeltra — CHANGES_REQUESTED** [review]
This is looking good. I just had a few comments inline. Let me know if you have any questions.

File: react-app/src/components/ResultSidebar.js
Comment (SenecaAnderson) [reply react-app/src/components/ResultSidebar.js]: I've moved the total message count. It uses the same CSS class as the total resource count text, so the color will update during rebasing.

File: react-app/src/components/ResultSidebar.js
Comment (jafeltra) [inline react-app/src/components/ResultSidebar.js:35]: Just writing down what we talked about in the working session, but I think we can remove the total count and just keep the total error and warning messages.

**Review by jafeltra — APPROVED** [review]
This looks good to me!

**Review by Dtphelan1 — APPROVED** [review]
This looks great. Still holding off on this until the other PR's are merged, but well done

**Pull request merged**
